src/main/java/com/google/devtools/build/lib/query2/cquery/ProtoOutputFormatterCallback.java

```diff
@@ -184,7 +206,35 @@ public void processOutput(Iterable<KeyedConfiguredTarget> partialResult)
       // logic in this next line. If this were to change (i.e. we manipulate targets any further),
       // we will want to add relevant tests.
       currentTarget = keyedConfiguredTarget;
-      builder.setTarget(formatter.toTargetProtoBuffer(accessor.getTarget(keyedConfiguredTarget)));
+      com.google.devtools.build.lib.packages.Target target = accessor.getTarget(
+          keyedConfiguredTarget);
+      Target.Builder targetBuilder = formatter.toTargetProtoBuffer(target);
+      if (target instanceof Rule && !Transitions.NONE.equals(options.transitions)) {
+        try {
+          for (TransitionResolver.ResolvedTransition resolvedTransition : transitionResolver.dependencies(
+              keyedConfiguredTarget)) {
+            if (resolvedTransition.options().isEmpty()) {
+              targetBuilder.getRuleBuilder().addConfiguredRuleInput(
+                  ConfiguredRuleInput.newBuilder().setLabel(resolvedTransition.label().toString()));
+            } else {
+              for (BuildOptions options : resolvedTransition.options()) {
+                BuildConfigurationEvent buildConfigurationEvent =
+                    getConfiguration(BuildConfigurationKey.withoutPlatformMapping(options)).toBuildEvent();
+                int configurationId = configurationCache.getId(buildConfigurationEvent);
+
+                targetBuilder.getRuleBuilder().addConfiguredRuleInput(
+                    ConfiguredRuleInput.newBuilder().setLabel(resolvedTransition.label().toString())
+                        .setConfigurationChecksum(options.checksum())
+                        .setConfigurationId(configurationId));
+              }
+            }
+          }
+        } catch (DependencyResolver.Failure | InconsistentAspectOrderException e) {
+          // This is an abuse of InterruptedException.
```

Comment:
This was copied and pasted from `TransitionsOutputFormatterCallback`. I think what it's saying is: `processOutput` only has two checked exceptions it's allowed to throw, `IOException` and `InterruptedException`, so rather than changing the `processOutput` API (a larger change), or throwing a `RuntimeException` (generally frowned upon in Java, but 🤷 seems better than this), hide the error in an `InterruptedException`.

I tried adding the exception to the interface, but ended up with a cyclic dependency because it ended up needing the `com.google.devtools.build.lib.cmdline.BatchCallback` interface (very high-level) to depend on the analysis package (very low level).

I could use a `RuntimeException`, or introduce a new Exception type, but I assume whoever originally wrote this code decided it was not worthwhile.

[This code](https://github.com/bazelbuild/bazel/blob/6594e0ca76305464f7f43ee236b21699ac4d3892/src/main/java/com/google/devtools/build/lib/query2/engine/OutputFormatterCallback.java#L45-L59) has an even sketchier hack, we could do something similar if we needed?

Reply:
ok thanks for the explanation. Looking at the previous commit, it seems to be part of a major change to reporting of Starlark errors and the call stack, so I would leave it.